kernel/ixgbe/src/lib.rs

```diff
@@ -1015,7 +992,10 @@ impl IxgbeNic {
         for rxq in rx_queues {            
             // the cpu id will tell which cache the data will need to be written to
             // TODO: choose a better default value
-            let cpu_id = rxq.cpu_id.unwrap_or(0) as u32;
+            let cpu_id = match rxq.cpu_id.into() {
+                Some(cpu_id) => cpu_id,
+                None => cpu::bootstrap_cpu().expect("Couldn't read BSP CpuId"),
+            }.into_u8() as u32;
```

Comment:
same comment, you can directly convert a CpuId into a u32, or just use the `.value()` function.

Or... wait. Does it need to be a CPU ID that fits within the bounds of a `u8`?

Reply:
Yes

Reply:
ah ok. we'll discuss on the other comment and wait for Ramla's response.

Reply:
yes, that field is only 8 bits.